tmail-backend/apps/memory/src/test/java/com/linagora/tmail/james/app/MemoryServerWithOpenPaasTest.java

```diff
@@ -0,0 +1,75 @@
+package com.linagora.tmail.james.app;
+
+import java.net.URI;
+import java.security.Permission;
+import java.util.Optional;
+import static org.apache.james.data.UsersRepositoryModuleChooser.Implementation.DEFAULT;
+import static org.assertj.core.api.AssertionsForClassTypes.fail;
+
+import org.apache.james.JamesServerBuilder;
+import org.apache.james.JamesServerExtension;
+import org.apache.james.utils.GuiceProbe;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.multibindings.Multibinder;
+import com.linagora.tmail.OpenPaasModuleChooserConfiguration;
+import com.linagora.tmail.configuration.OpenPaasConfiguration;
+import com.linagora.tmail.encrypted.MailboxConfiguration;
+import com.linagora.tmail.encrypted.MailboxManagerClassProbe;
+import com.linagora.tmail.module.LinagoraTestJMAPServerModule;
+
+public class MemoryServerWithOpenPaasTest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MemoryServerWithOpenPaasTest.class);
+
+    static JamesServerExtension jamesServerExtension = new JamesServerBuilder<MemoryConfiguration>(tmpDir ->
+        MemoryConfiguration.builder()
+            .workingDirectory(tmpDir)
+            .configurationFromClasspath()
+            .mailbox(new MailboxConfiguration(false))
+            .usersRepository(DEFAULT)
+            .openPaasModuleChooserConfiguration(OpenPaasModuleChooserConfiguration.ENABLED)
+            .build())
+        .lifeCycle(JamesServerExtension.Lifecycle.PER_CLASS)
+        .server(configuration -> MemoryServer.createServer(configuration)
+            .overrideWith(new LinagoraTestJMAPServerModule())
+            .overrideWith(binder -> Multibinder.newSetBinder(binder, GuiceProbe.class).addBinding().to(MailboxManagerClassProbe.class))
+            .overrideWith(binder -> binder.bind(OpenPaasConfiguration.class).toInstance(
+                new OpenPaasConfiguration(
+                    Optional.empty(),
+                    URI.create("https://127.0.0.1"),
+                    "admin",
+                    "admin"
+                )
+            )))
+        .build();
+
+    @BeforeAll
+    static void setUp() {
+        System.setSecurityManager(new NoExitSecurityManager());
+    }
```

Comment:
WHY?

Reply:
On failure, the server does not throw an exception it exits by calling `System.exit`. Here we are trying to test that the memory server fails when the OpenPaas module is enabled and there is no rabbitmq channel pool to fallback to.

Reply:
Here is is worth a comment?